src/netfilter.rs

```diff
@@ -0,0 +1,285 @@
+//! Netfilter protocols
+//!
+//! Protocols used for communicating with netfilter. Currently, this contains (partial) support for
+//! NFLOG, NFQUEUE and CONNTRACK will be added later.
+//!
+//! See the examples in the git repository for actual, working code.
+
+use std::ffi::CString;
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+use libc::c_int;
+
+use crate::consts::netfilter::{LogCopyMode, NfLogAttr, NfLogCfg};
+use crate::err::{DeError, SerError};
+use crate::nlattr::Nlattr;
+use crate::{Nl, StreamReadBuffer, StreamWriteBuffer};
+
+type Nlattrs = Vec<Nlattr<NfLogAttr, Vec<u8>>>;
+
+#[derive(Copy, Clone, Debug)]
+struct Timestamp {
+    secs: u64,
+    usecs: u64,
+}
+
+impl Nl for Timestamp {
+    fn serialize(&self, m: &mut StreamWriteBuffer) -> Result<(), SerError> {
+        u64::to_be(self.secs).serialize(m)?;
```

Comment:
Are you positive these are all in big endian? If you are assuming that because netlink is a network protocol, netlink always uses the endianness of the host system, whether big or little, as it is exclusively a local protocol. If you have an example of this protocol specifically being big endian values, can you provide that evidence? [This comment](https://chromium.googlesource.com/native_client/linux-headers-for-nacl/+/2dc04f8190a54defc0d59e693fa6cff3e8a916a9/include/linux/netlink.h#140) was how I originally discovered this as netlink will explicitly let you know when it's network byte order as opposed to host byte order.

Reply:
I'm positive. First, the type in the headers are marked as `__be32`.

Second, I did it without these conversions first and it didn't work ‒ the values for timestamp, for example, usually produced overflows when converting to SystemTime. When I do the conversion, it produces sane values.

Reply:
Excellent. This looks good then. I know it can get a little tedious providing evidence so thank you for bearing with me.

Reply:
That's fine. I take a thorough review as a seal of quality of the library :-).